tiledb/sm/storage_manager/storage_manager.cc

```diff
@@ -701,6 +803,110 @@ Status StorageManager::object_move(
   return vfs_->move_dir(old_uri, new_uri);
 }
 
+Status StorageManager::get_fragment_info(
+    const ArraySchema* array_schema,
+    uint64_t timestamp,
+    const EncryptionKey& encryption_key,
+    std::vector<FragmentInfo>* fragment_info) {
+  fragment_info->clear();
+
+  // Get fragment URIs
+  std::vector<URI> fragment_uris;
+  RETURN_NOT_OK(get_fragment_uris(array_schema->array_uri(), &fragment_uris));
+
+  // Check if the array is empty
+  if (fragment_uris.empty())
+    return Status::Ok();
+
+  // Sort the URIs by timestamp
+  std::vector<std::pair<uint64_t, URI>> sorted_fragment_uris;
+  get_sorted_fragment_uris(fragment_uris, timestamp, &sorted_fragment_uris);
+
+  // Get the rest of fragment info
+  uint64_t domain_size = 2 * array_schema->coords_size();
+  uint64_t size;
+  bool sparse, in_cache;
+  for (const auto& uri : sorted_fragment_uris) {
+    // Determine if the fragment is sparse
+    URI coords_uri =
+        uri.second.join_path(constants::coords + constants::file_suffix);
+    RETURN_NOT_OK(vfs_->is_file(coords_uri, &sparse));
+
+    // Get fragment size
+    RETURN_NOT_OK(vfs_->dir_size(uri.second, &size));
+
+    // Get fragment non-empty domain
+    auto metadata =
```

Comment:
Don't need `new` and `delete` for this local object -- can just do `FragmentMetadata metadata(...);` to stack allocate.